src/daemons/rhsmcertd.c

```diff
@@ -748,10 +868,28 @@ main (int argc, char *argv[])
         return EXIT_FAILURE;
 
     if (get_lock () != 0) {
-        error ("unable to get lock, exiting");
+        error ("Unable to get lock, exiting");
         return EXIT_FAILURE;
     }
 
+    // NOTE: it is important to create callback function for signals after calling daemon()
+    // Create main context for main loop
+    GMainContext *main_context;
+    main_context = g_main_context_new ();
```

Comment:
i think you can use the existing context, ie `g_main_context_default()`

Reply:
Yeah, that is true. We needn't to create another new context.